faunadb-java-dsl/src/main/java/com/faunadb/client/types/Creators.java

```diff
@@ -0,0 +1,196 @@
+package com.faunadb.client.types;
+
+import com.faunadb.client.errors.FaunaException;
+import com.faunadb.client.types.Properties.Property;
+import com.faunadb.client.types.Types.SimpleType;
+import com.faunadb.client.types.Value.ObjectV;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.*;
+
+import static com.faunadb.client.types.Decoder.decodeImpl;
+import static java.lang.String.format;
+
+class Creators {
+    interface Creator {
+        Object newInstance(ObjectV value) throws InvocationTargetException, IllegalAccessException, InstantiationException;
```

Comment:
It doesn't make sense to make it generic since the context in which it's called doesn't have type information, it's called during decoding through reflection